repos/c2sm/packages/icon/package.py

```diff
@@ -611,19 +638,46 @@ def configure_args(self):
                     for d in self._get_cuda_ccbin_link_paths(gcc, 'libstdc++')
                 ]
                 cuda_host_compiler_stdcxx_libs.append('-lstdc++')
+                config_vars['NVCFLAGS'].extend(
+                    ['-ccbin {0}'.format(cuda_host_compiler)])
             else:
                 cuda_host_compiler = self.compiler.cxx
                 cuda_host_compiler_stdcxx_libs = self.compiler.stdcxx_libs
 
-            config_vars['NVCFLAGS'].extend([
-                '-ccbin {0}'.format(cuda_host_compiler), '-g', '-O3',
-                '-arch=sm_{0}'.format(gpu)
-            ])
+            config_vars['NVCFLAGS'].extend(
+                ['-g', '-O3', '-arch=sm_{0}'.format(gpu)])
```

Comment:
Here you break the variant `+cuda-gcc`. Can you refactor in a way this only happens with `+dsl`?